Test restarting worker in Travis tests

bkgoksel opened the pull request:
This is to make sure worker restarts are fine and there is no
serialization/deserialization problems in worker state committing on
disk.

Commit:
Test restarting worker in TRAVIS_TAG

This is to make sure worker restarts are fine and there is no
serialization/deserialization problems in worker state committing on
disk.

percyliang reviewed (commented):
What problem is this reacting to?

bkgoksel commented:
Not a specific problem, but while working on #1505 I realized I introduced some serialization/deserialization bugs that were not caught by the tests and I only caught them after doing this simple worker restart.  I also added some unit tests after that, but I think adding an end-to-end state serialization/deserialization test by doing this restart wouldn't hurt.

Commit:
Review edits

Commit:
Merge branch 'master' into worker-restart-test
